UI - show table/view detail #1996

maheshsapkal opened the pull request.

rymurr reviewed (commented):
Thanks @maheshsapkal this looks good.

Could you please:
1) post screenshots summarizing the changes
2) fix the todos in tests
3) don't use linter skip comments (or add a comment to justify why they are being used)

maheshsapkal commented:
@rymurr please find screenshots below for content view, for different types.
![DeltaLakeContents](https://user-images.githubusercontent.com/91591279/141118529-335a54de-3955-4df0-af72-9bda54980111.png)
![IcebergContent](https://user-images.githubusercontent.com/91591279/141118537-5d605141-6f8f-457a-9f8e-01500e74fdef.png)
![SqlViewContent](https://user-images.githubusercontent.com/91591279/141118542-4b8095a9-7240-43ba-9f51-531f2c7377c9.png)

maheshsapkal commented:
> Thanks @maheshsapkal this looks good.
> 
> Could you please:
> 
> 1. post screenshots summarizing the changes
> 2. fix the todos in tests
> 3. don't use linter skip comments (or add a comment to justify why they are being used)

@rymurr 
1. Please find the above screenshot.
2. Fixed todo in tests
3. Fixed typescript error and removed linter skip comments.

ajantha-bhat commented on ui/src/Explore/Components/ContentsView.test.tsx:
```diff
@@ -0,0 +1,140 @@
+  type: "ICEBERG_TABLE",
+  id: "5d8bdfec-96d7-44f6-8770-1ffcf09e044d",
+  metadataLocation: "/path/to/metadata/",
+  idGenerators: "6c3563f6847feda9",
```
Also as per new release `idGenerators` is removed. 

please use the latest ICEBERG_TABLE content.

ajantha-bhat replied:
https://github.com/projectnessie/nessie/pull/2626

maheshsapkal replied:
@ajantha-bhat Looks like the following fields are newly added, do we have to show them on UI? it seems like those are internal purposes only but wanted to make sure.
 schemaId, specId, sortOrderId
 cc: @nastra

nastra replied:
let's omit them for now. We can always add them later to the UI if necessary

maheshsapkal replied:
@ajantha-bhat idGenerators is removed from test cases. Also updated latest content for same.

nastra commented on ui/src/Explore/Components/ContentView.tsx:
```diff
@@ -0,0 +1,157 @@
+    return <EmptyMessageView message="Invalid type" />;
+  };
+
+  const renderContents = () => {
```
renderContents -> renderContent

nastra commented on ui/src/Explore/Components/ContentView.tsx:
```diff
@@ -0,0 +1,157 @@
+  };
+
+  const renderContents = () => {
+    switch (tableContents.type) {
```
tableContents -> tableContent

nastra commented on ui/src/Explore/Components/ContentView.tsx:
```diff
@@ -0,0 +1,157 @@
+    );
+  };
+
+  const renderView = (sqlViewContents: SqlView) => {
```
sqlViewContents -> sqlViewContent

nastra commented on ui/src/Explore/Components/ContentView.tsx:
```diff
@@ -0,0 +1,157 @@
+    );
+  };
+
+  const renderIcebergView = (icebergContents: IcebergTable) => {
```
icebergContents -> icebergContent

nastra commented on ui/src/Explore/Components/ContentView.tsx:
```diff
@@ -0,0 +1,157 @@
+    );
+  };
+
+  const renderDeltaView = (deltaContents: DeltaLakeTable) => {
```
deltaContents -> deltaContent

nastra commented on ui/src/Explore/Components/ContentView.test.tsx:
```diff
@@ -0,0 +1,142 @@
+  dialect: Dialect.Hive,
+};
+
+const contentsList = {
```
contentsList -> contentList

nastra commented on ui/src/Explore/TableListing.tsx:
```diff
@@ -67,51 +80,120 @@ const entryToKey = (entry: Entry): Key => {
   return {
     name: entry.name.elements[entry.name.elements.length - 1],
     type: entry.type === "UNKNOWN" ? "CONTAINER" : "TABLE",
+    contentsKey: entry.name,
```
contentsKey -> contentKey

nastra commented on ui/src/Explore/TableListing.tsx:
```diff
@@ -67,51 +80,120 @@ const entryToKey = (entry: Entry): Key => {
 interface Key {
   name: string;
   type: "CONTAINER" | "TABLE";
+  contentsKey: ContentKey;
```
contentsKey -> contentKey

nastra reviewed (commented):
still a few renames that should be done

maheshsapkal commented on ui/src/Explore/Components/ContentsView.tsx:
```diff
@@ -0,0 +1,184 @@
+            <span>Type: </span>
+            <span className={"ml-2"}> {tableContents.type}</span>
+          </div>
+          {id && (
```
> still a few renames that should be done

Done

nastra reviewed (approved):
LGTM
